core/math/math_funcs.cpp

```diff
@@ -30,30 +30,29 @@
 
 #include "math_funcs.h"
 
-#include "core/os/os.h"
-
-pcg32_random_t Math::default_pcg = { 12047754176567800795ULL, PCG_DEFAULT_INC_64 };
+RandomBase Math::default_rand(RandomBase::DEFAULT_SEED, RandomBase::DEFAULT_INC);
 
 #define PHI 0x9e3779b9
 
 // TODO: we should eventually expose pcg.inc too
 uint32_t Math::rand_from_seed(uint64_t *seed) {
+	// TODO: create a RandomBase instance
```

Comment:
I think something like this might work (untested):
```cpp
RandomBase rng = RandomBase(*seed, RandomBase::DEFAULT_INC);
uint32_t r = rng.rand();
*seed = rng.get_seed();
return r;
```

Reply:
It is the same ! Changed..

Reply:
I think you can remove the other TODO above, that's basically what this PR does :)